tools/reach-calculator.html

```diff
@@ -0,0 +1,233 @@
+        </div>
+
+        <div id="apiTab" class="optionTab">
+            <label>API/feature usage share (%):<br>
+            <input type="number" id="apiUsageInput" min="0" max="100"></label>
+            <div>Note: <a href="https://chromestatus.com/metrics/feature/popularity" target="_blank">Chrome Platform Status</a> is useful for JS APIs.</div>
+        </div>
+
+        <div>
+            <label><abbr title="Further reduction assumption, e.g. visiting 2+ pages, particular version, etc.">[Optional] Reduction assumption (%):<br>
+            <input type="number" id="reductionFactorInput" min="0" max="100"></abbr></label><br>
+        </div>
+
+        <div>
+            <label for="confidenceInput">Additional unaccounted-for breakage?</label><br>
+            <label><input type="radio" id="low" name="confidenceInput" value="low" checked> Low confidence</label><br>
+            <label><input type="radio" id="medium" name="confidenceInput" value="medium"> Medium confidence</label><br>
+            <label><input type="radio" id="high" name="confidenceInput" value="high"> High confidence</label><br>
+        </div>
+    
+        <hr>
+        <div id="results"></div>
+        <div>(Use this and the Severity Assessment Matrix to get a priority level.)</div>
+    </div>
+</div>
+
+<script>
+(function () {
+    'use strict';
+
+    // Multipliers for confidence that the breakage issue is more widespread.
+    const confidenceLevels = {
+        low: 1,
+        medium: 10,
+        high: 100
+    };
+
+    // Format results for display so they're between 0 and 100.
+    function formatResult (num) {
+        if (num === 0 || isNaN(num)) {
+            return 0;
+        } else if (num > 1) {
+            return 100;
+        } else {
+            num *= 100;
+            return num.toPrecision(3);
+        }
+    }
+
+    // Hide/display the domain rank and API usage fields.
+    function toggleTab (type) {
+        document.body.className = type;
+        if (type === 'domain') {
+            apiUsageInput.value = '';
+        } else {
+            domainRankInput.value = '';
+        }
+        doCalculation();
+    }
+
+    // Calculate breakage reach as a percentage of page loads.
+    function doCalculation () {
+        const domainRank = domainRankInput.value;
+        const apiUsage = apiUsageInput.value;
+        const reductionFactor = reductionFactorInput.value;
+        const confidence = document.querySelector('input[name="confidenceInput"]:checked').value;
+        let result = '';
+
+        // Domain popularity is calculated using a trend line formula based on rank:pageload data from https://dl.acm.org/doi/10.1145/3517745.3561418
+        // Domain rank 1 is an outlier and therefore an exception to the formula.
+        let domainPopularity = 0;
+        if (domainRank) {
+            domainPopularity = (domainRank > 1) ? (0.0687189317758425 * Math.pow(domainRank, -1.03435633987409)) : 0.17;
```

Comment:
Are these constants from the paper? I did wonder how you'd done this 👍🏻

Reply:
Yes, although I had to reverse engineer it a bit. They only reveal the following data:

* The top site has 17% of traffic
* The top 6 sites have 25% of traffic
* The top 100 sites have 40% of traffic
* The top 10K sites have 70% of traffic
* The top 1M sites have 95%+ of traffic

So I worked backwards to get a rough long tail of traffic for each domain rank, and then used a trend line to get the constants.